clients/web/src/views/widgets/MetadataWidget.js

```diff
@@ -2,80 +2,418 @@
+            },
+            validation: {
+                from: {
+                    simple: [
+                        function (value) {
+                            try {
+                                var jsonValue = JSON.parse(value);
+                                /* This may succeed when we don't want it to (for instance with the
+                                 * value 'false' or '1234'), so check and only switch to JSON if we
+                                 * got an object back. */
+                                if (jsonValue && typeof jsonValue === 'object' && jsonValue !== null) {
+                                    return true;
+                                }
+                            } catch (e) {}
+
+                            return false;
+                        },
+                        'The simple field is not valid JSON and can not be converted.'
+                    ]
+                }
+            },
+            template: girder.templates.jsonMetadatumView
+        }
+    },
+
     events: {
-        'click .g-widget-metadata-add-button': 'addMetadata',
-        'click .g-widget-metadata-edit-button': 'editMetadata'
+        'click .g-add-json-metadata': function (event) {
+            this.addMetadata(event, 'json');
+        },
+        'click .g-add-simple-metadata': function (event) {
+            this.addMetadata(event, 'simple');
+        }
+    },
+
+    initialize: function (settings) {
+        this.item = settings.item;
+        this.accessLevel = settings.accessLevel;
+        this.item.on('g:changed', function () {
+            this.render();
+        }, this);
+        this.render();
+    },
+
+    setItem: function (item) {
+        this.item = item;
+        return this;
     },
 
-    addMetadata: function () {
+    // Does not support modal editing
+    getModeFromValue: function (value) {
+        return (typeof value === 'object') ? 'json' : 'simple';
+    },
+
+    addMetadata: function (event, mode) {
+        var EditWidget = this.modes[mode].editor;
         var newRow = $('<div>').attr({
             class: 'g-widget-metadata-row editing'
         }).appendTo(this.$el.find('.g-widget-metadata-container'));
-        this.metadatumEditWidget = new girder.views.MetadatumEditWidget({
+
+        var widget = new girder.views.MetadatumWidget({
             el: newRow,
+            mode: mode,
+            key: '',
+            value: '',
+            item: this.item,
+            accessLevel: this.accessLevel,
+            girder: girder,
+            parentView: this
+        });
+
+        var newEditRow = widget.$el.append('<div></div>');
+
+        new EditWidget({
+            el: newEditRow.find('div'),
             item: this.item,
             key: '',
             value: '',
+            accessLevel: this.accessLevel,
             newDatum: true,
+            parentView: widget
+        }).render();
+    },
+
+    render: function () {
+        var _this = this;
```

Comment:
`_this` isn't necessary here, if you pass a third argument to `_.each`, it will be used as the binding context on the internal function :smile: